src/riak_kv_bucket.erl

```diff
@@ -43,6 +44,43 @@
 
 -export_type([props/0]).
 
+-define(VALID_PROPERTIES,
+        ["allow_mult",
+         "basic_quorum",
+         "big_vclock",
+         "bucket_type",
+         "chash_keyfun",
+         "dvv_enabled",
+         "dw",
+         "last_write_wins",
+         "linkfun",
+         "n_val",
+         "notfound_ok",
+         "old_vclock",
+         "postcommit",
+         "pr",
+         "precommit",
+         "pw",
+         "r",
+         "rw",
+         "small_vclock",
+         "w",
+         "write_once",
+         "young_vclock",
+         %% TS-specific ones:
+         "ddl",
+         "table_def"
+        ]).
+
+-spec is_valid_property(string() | binary()) -> boolean().
```

Comment:
Understood, but we still need to remove all of this. Catch the badarg if you need to.